aten/src/ATen/CPUGenerator.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <ATen/core/Generator.h>
+#include <ATen/core/PhiloxRNGEngine.h>
+
+namespace at {
+
+struct CPUGenerator;
+
+namespace detail {
+
+// Global generator state and constants
+static std::once_flag cpu_device_flag;
+static std::unique_ptr<CPUGenerator> default_gen_cpu;
```

Comment:
This looks wrong, you shouldn't declare variables in headers. Move these to the cpp file.